source/pdal/pdalc_pipeline.cpp

```diff
@@ -167,25 +166,25 @@ extern "C"
         if (pipeline && schema && size > 0)
         {
             Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
-            pdal::PipelineExecutor *executor = ptr->get();
+
             schema[0] =  '\0';
             schema[size - 1] = '\0';
 
-            if (executor)
+            try
             {
-                try
-                {
-                    std::string s = executor->getSchema();
-                    std::strncpy(schema, s.c_str(), size);
-                    result = std::min(s.length(), size);
-                }
-                catch (const std::exception &e)
-                {
-                    printf("Found error while retrieving pipeline's schema: %s\n", e.what());
-                }
+                std::stringstream strm;
+                MetadataNode meta = ptr->manager->pointTable().layout()->toMetadata();
+                MetadataNode root = meta.clone("schema");
+                pdal::Utils::toJSON(root, strm);
+                std::strncpy(schema, strm.str().c_str(), size);
+                result = std::min(strm.str().length(), size);
```

Comment:
Again, it's confusing as to whether you're guaranteeing a null-terminated string or not.

Reply:
see comment below